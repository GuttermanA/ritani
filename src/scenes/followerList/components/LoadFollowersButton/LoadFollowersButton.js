import React, { Component } from "react";
import { Button } from "components";

class LoadFollowersButton extends Component {
  // shouldComponentUpdate(nextProps) {
  //   return nextProps.disabled !== this.props.disabled ? true : false;
  // }
  render() {
    const { remainingPages } = this.props;
    console.log(this.props);
    const content = remainingPages ? "Load" : "End of Followers";
    return (
      <Button className="button centered" disabled={!remainingPages}>
        {content}
      </Button>
    );
  }
}

export default LoadFollowersButton;
