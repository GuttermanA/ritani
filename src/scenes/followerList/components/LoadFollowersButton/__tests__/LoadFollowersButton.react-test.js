import React from "react";
import renderer from "react-test-renderer";
import LoadFollowersButton from "../LoadFollowersButton";

describe("<LoadFollowersButton/>", () => {
  it("renders correctly", () => {
    const component = renderer.create(<LoadFollowersButton />);
    expect(component).toMatchSnapshot();
  });
});
