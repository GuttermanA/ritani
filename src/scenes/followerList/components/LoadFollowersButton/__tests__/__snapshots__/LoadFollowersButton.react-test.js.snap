// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadFollowersButton/> renders correctly 1`] = `
Array [
  <button
    className="button centered"
    type="submit"
  >
    Load
  </button>,
  <div />,
]
`;
